dll/win32/syssetup/install.c

```diff
@@ -524,24 +526,177 @@ EnableUserModePnpManager(VOID)
     return bRet;
 }
 
+static BOOL
+RegisterDevNotificationForHwnd(
+    IN HWND hWnd,
+    OUT HDEVNOTIFY* hDeviceNotify)
+{
+    DEV_BROADCAST_DEVICEINTERFACE NotificationFilter = { 0 };
+    NotificationFilter.dbcc_size = sizeof(DEV_BROADCAST_DEVICEINTERFACE);
+    NotificationFilter.dbcc_devicetype = DBT_DEVTYP_DEVICEINTERFACE;
+
+    *hDeviceNotify = RegisterDeviceNotification(hWnd,
+                                                &NotificationFilter,
+                                                DEVICE_NOTIFY_WINDOW_HANDLE | DEVICE_NOTIFY_ALL_INTERFACE_CLASSES);
+
+    return *hDeviceNotify != NULL;
+}
+
+static BOOLEAN
+GetFriendlyDeviceName(
+    IN PDEV_BROADCAST_DEVICEINTERFACE pDevice,
+    OUT LPCWSTR* pszName)
+{
+    BOOLEAN result = FALSE, DeviceInterfaceDataInitialized = FALSE;
+    HDEVINFO hList = NULL;
+    SP_DEVICE_INTERFACE_DATA DeviceInterfaceData = { 0 };
+
+    if (!pDevice
+        || ~pDevice->dbcc_devicetype & DBT_DEVTYP_DEVICEINTERFACE
```

Comment:
`!(pDevice->dbcc_devicetype & DBT_DEVTYP_DEVICEINTERFACE)` is a more common way to write this